✨ Changing the <h1> tags into MudText h1 tags

Sean-Stilwell opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->

## Description
<!-- Please provide a brief description of the changes made in this pull request -->

This PR makes the `<h1>` tags into `<MudText Typo="Typo.h1">` tags.

Where needed, a `<span>` has been added to hold the main content identifiers.

### Before

![image](https://github.com/user-attachments/assets/8d28ac86-8e19-43c4-a533-556af9eee2af)

### After

![image](https://github.com/user-attachments/assets/c20c48f8-65d3-420d-8d24-e3ecf2a081b3)

## How Has This Been Tested?
<!-- Please describe how you tested your changes. -->
<!-- Have you created new tests or updated existing ones? -->
<!-- e.g. unit | visually | none -->

Locally validated the changes

## Types of changes
<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Unit test (new or update to tests)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (change that would cause documentation to be updated)
- [ ] Configuration change (change that would affect the configuration of the application)

<!-- If you made any visual changes, provide screenshots of before/after, it its moving parts, please provide high quality gif, wemb or mp4 -->

## Checklist:
<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask! -->
- [x] PR is submitted to the correct branch `develop`.
- [x] Code follows the code style of this project.
- [x] Changes are covered by Specflow tests and all new or existing tests are passing.
- [x] Any new or updated translatable strings have been added to the localization files.
- [x] Necessary and relevant documentation has been created or updated.
- [x] There are either no changes to the application's configuration or the necessary changes in the infrastructure repository are included in a separate PR.

<!-- Link to the "infrastructure" repository PR here -->

Sean-Stilwell committed:
Updating the landing page to use MudBlazor h1 styling

Sean-Stilwell committed:
Updating the account setup page to use MudBlazor h1 styling

Sean-Stilwell committed:
Updating the workspace dashboard page to use MudBlazor h1 styling

Sean-Stilwell committed:
Updating the terms page to use MudBlazor h1 styling

Sean-Stilwell committed:
Updating the news page to use MudBlazor h1 styling

Sean-Stilwell committed:
Updating the workspace reports page to use MudBlazor h1 styling

Sean-Stilwell committed:
Updating the home page to use MudBlazor h1 styling

Sean-Stilwell committed:
Updating the account pages to use MudBlazor h1 styling

Sean-Stilwell committed:
Updating the workspace metadata to use MudBlazor h1 styling

Sean-Stilwell committed:
Updating the explore page to use MudBlazor h1 styling

Sean-Stilwell committed:
Updating achievement, help, and resource page

Sean-Stilwell committed:
Updating various workspace pages

Sean-Stilwell committed:
Updating various workspace pages and public profile

Sean-Stilwell committed:
Adding the rest of the workspace pages

Sean-Stilwell committed:
Minor formatting changes on main page

yjmrobert reviewed (changes requested):
Can we refactor these changes into a component we can reuse?

Something like `DatahubMainContentTitle` or some variant around that (not `PageTitle` as that's reserved for the browser tab)

Sean-Stilwell committed:
Switching it all to use a component

Sean-Stilwell committed:
Making terms and conditions page open in a new tab

yjmrobert committed:
🔧 Updated parameter type to `RenderFragment`

yjmrobert committed:
🔧 Refactor main content title component usage

Replaced HTML spans with DHMainContentTitle component across multiple Razor pages for consistency. Updated DHMainContentTitle implementation to support both ChildContent and string Title parameters with validation.

yjmrobert reviewed (approved):
Nice work! I added a `ChildContent` to the component along with some parameter checks so it could be used across the whole portal
